script.js

```diff
@@ -1,4 +1,71 @@
-var main = function (input) {
-  var myOutputValue = 'hello world';
-  return myOutputValue;
-};
+
+var diceRoll = function(){
+  return Math.floor(Math.random()*6+1)
+}
+
+// score leaderboard tracker game with 4 players//
+var numOfPlayers = 1000
+var scoreArray = [0]
+var player = 1 
+var numOfPlayersSpecified = false
+var diceOne = 0
+var diceTwo = 0
+//var main = function(input,input1)
+var main = function(input){
+  console.log(input)
+  // take the user input as the number of players in the game
+  if (numOfPlayersSpecified==false){
+    // input validation
+    if (input <2 || input >10){
+      return `Greetings! Please indicate the number of players`
+    }
+    numOfPlayers = input;
+    numOfPlayersSpecified = true
+    return "Starting Game. It is Player 1's turn. Press Submit to Roll"
+  }
+  // player playing is same as number of players restart player count to 1
+  if (player == Number(numOfPlayers) + 1){
+    player = 1
+  }
+  // roll dice and determine dice number for each player
+  diceOne = diceRoll()
+  diceTwo = diceRoll()
+  diceNumber = determineDiceNumber(diceOne,diceTwo)
```

Comment:
Good use of arguments in functions because it shows you know how they work! One thing to note is, since diceOne and diceTwo are global variables, they can be accessed directly from any function.